docs/proposals/mvp-dashboard-design.md

```diff
@@ -43,18 +43,125 @@ can query the relevant test builds.
 
 ## Proposal
 
-![Diagram 1](../images/perf-dash-proposal-diagram.png)
+![alt text](../images/perf-dash-proposal-diagram.png "Perf-Dash Diagram")
+
+### Server
+
+The system we propose requires the development of a server API that will be
+used to serve queries that a user makes via the dashboard. To speed up the
+initial development process, these queries will be hardcoded for test results.
+Adding query build functionality for the user is a feature that will be
+developed in future development cycles.
+
+When the server receives input from the user (a query), the server will send
+these queries to the data storage.  The data storage returns the filtered
+performance test results. These results are then pushed to the dashboard.
+The server expects a json result from the data store. The format and structure
+of this json response is a task that will need to be completed during the
+development of the perf-dash.
+
+### Dashboard
+
+The dashboard serves the purpose of allowing the user to make queries of the
+results stored in the data storage unit. The dashboard provides a simple user
+interface with a list of drop down menus for options such as: lab, config,
+test, run, and metric. The dashboard communicates with the server (both
+running on the same node) using an NGINX reverse proxy.
+
+The query is made via HTML requests in the form of `GET /api/labs`,
+`GET /api/labs/{lab id}`. The expected result is a json with the appropriate
+test data. This json is then used to create the graph of the specified metrics
+as they trend over past builds.
 
 ## Rationale
 
-A discussion of alternate approaches and the trade offs, advantages, and
-disadvantages of the specified approach.
+### Choice to build our own front-end
+
+1. **Avoid Grafana learning curve**
```

Comment:
IMHO, I don't think this is a solid argument.

Grafana really is not that hard to pick up. The main argument I see is that Grafana being a generic dashboard meant for services and systems monitoring & visualization, not really for benchmark and performance tests comparisons or the domain concepts we want to model with our proposed system. Because of this. it hurts discoverability of available workloads results, and environments, as well as, it would be hard for other users to even know which data sets were executed together or are even relevant since it will simply look as continuous time series plot where all environments are combined (if plotted naively).